subsys/bluetooth/controller/Kconfig

```diff
@@ -128,6 +128,15 @@ config BT_CTLR_TX_BUFFERS
 	  packets can be enqueued, with 18 packets with L2CAP payload size of 1
 	  byte can be acknowledged.
 
+config BT_CTLR_TX_CTRL_BUFFERS
+	int "Number of Tx Control buffers"
+	default 2
+	range 1 19
```

Comment:
Curious about the number 19?

Reply:
the current value of 2 that is upstream fails under race conditions. Some control procedure enqueue 2 ctrl PDUs simultaneously, example, ENC_RSP and START_ENC_REQ, at probably a prior local initiated ctrl PDU could be in queue. leading to buffer underrun assertion. Is this PR a fix for this scenario?

Reply:
No good reason for 19 (just the same max as for BT_CTLR_TX_BUFFERS). And yes, this change was triggered by observing scenarios where we ran out of buffers (4 buffers were enough for those scenarios).

Reply:
Event n+0:
M -> S: Enc Req PDU
M <- S: Slave initiated Ctrl Req PDU (Buffer 1)

Event n+1:
M -> S: Empty PDU
S -> M: Slave initiated Ctrl Req PDU (Buffer 1)

Event n+2:
M -> S: Empty PDU
S -> M: ...

To be continued... need to be away

Reply:
The scenario where I see the failure has several masters connect to the a slave almost at the same time. As far as I understand, your reasoning above would apply to every connection and in the worst case 4*n buffers would be needed. That seems a bit excessive, but I would not know where to draw the line in the general case.

Reply:
The problem with running out of TX control buffers actually manifests as running out of space elsewhere (either lacking space in the `done` queue or exhausting the `pdu_rx_free` buffers). If `rx_demux_rx` returns -ENOBUFS due to lack of TX control buffers, the loop over `memq_ull_rx` in `rx_demux` exits. Other buffers on their way up are therefore stuck at that point, leading other pools to be exhausted. If we want to get away with allocating fewer TX control buffers than the worst case, we need to find a way to keep buffers flowing.